legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/dbExtension.pure

```diff
@@ -82,6 +82,12 @@ Class meta::relational::functions::sqlQueryToString::DbConfig
      $this.dbExtension.selectSQLQueryProcessor->eval($s, $sgc, $isSubSelect);
    }: String[1];
 
+   upsertSQLQueryProcessor(u:UpsertSQLQuery[1], sgc:SqlGenerationContext[1])
+   {
+     assert($this.dbExtension.upsertSQLQueryProcessor->isNotEmpty(), '[unsupported-api] Upsert operation not supported for Database Type: ' + $this.dbType->toString());
+     $this.dbExtension.upsertSQLQueryProcessor->toOne()->eval($u, $sgc);
```

Comment:
You could avoid the assert call, and just pass a message to the `toOne` function.

`$this.dbExtension.upsertSQLQueryProcessor->toOne('[unsupported-api] Upsert operation not supported for Database Type: ' + $this.dbType->toString())->eval($u, $sgc);`

If the toOne will throw given the effective multiplicity is not 1, then the message is what will pop up as the error  message.

Reply:
Thanks for this. This is great.